File: app/back/src/main/java/net/seijishikin/jp/normalize/shuushi_doc/v05/constants/ShuushiDocV5YoushikiEdaOutcomeKeihiConstants.java
package net.seijishikin.jp.normalize.shuushi_doc.v05.constants;

/**
 * 政治資金収支報告書V5様式枝区分項目支出経費
 */
public final class ShuushiDocV5YoushikiEdaOutcomeKeihiConstants {

    /** 人件費 */
    public static final short JINKENHI = 1;

    /** 光熱水費 */
    public static final short KOUNETSUSUI = 2;

    /** 備品・消耗品費 */
    public static final short BIHIN_SHOUMOUHIN = 3;

    /** 事務所費 */
    public static final short JIMUSHO = 4;

    /** 人件費 */
    public static final String JINKENHI_TEXT = "人件費";

    /** 光熱水費 */
    public static final String KOUNETSUSUI_TEXT = "光熱水費";

    /** 備品・消耗品費 */
    public static final String BIHIN_SHOUMOUHIN_TEXT = "備品・消耗品費";

    /** 事務所費 */
    public static final String JIMUSHO_TEXT = "事務所費";

    private ShuushiDocV5YoushikiEdaOutcomeKeihiConstants() {

    }

    /**
     * 表示ラベルを取得する
     *
     * @param keyValue 設定値
     * @return 表示テキスト
     * @throws IllegalArgumentException このクラスで使用する定数でない場合
     */
    public static final String getLabel(final short keyValue) throws IllegalArgumentException { // NOPMD UnChecked

        switch (keyValue) {

            case JINKENHI:
                return JINKENHI_TEXT;

            case KOUNETSUSUI:
                return KOUNETSUSUI_TEXT;

            case BIHIN_SHOUMOUHIN:
                return BIHIN_SHOUMOUHIN_TEXT;

            case JIMUSHO:
                return JIMUSHO_TEXT;

            default:
                throw new IllegalArgumentException("Unexpected value: " + keyValue);
        }
    }

}

File: app/seijishikin-jp-normalize/src/components/main/constants/shuushi_doc/v05/shuushiDocV5YoushikiEdaOutcomeKeihiConstants.ts
export class ShuushiDocV5YoushikiEdaOutcomeKeihiConstants {

    /** 人件費 */
    static readonly JINKENHI = 1;

    /** 光熱水費 */
    static readonly KOUNETSUSUI = 2;

    /** 備品・消耗品費 */
    static readonly BIHIN_SHOUMOUHIN = 3;

    /** 事務所費 */
    static readonly JIMUSHO = 4;

    /** 人件費 */
    static readonly JINKENHI_TEXT = "人件費";

    /** 光熱水費 */
    static readonly KOUNETSUSUI_TEXT = "光熱水費";

    /** 備品・消耗品費 */
    static readonly BIHIN_SHOUMOUHIN_TEXT = "備品・消耗品費";

    /** 事務所費 */
    static readonly JIMUSHO_TEXT = "事務所費";

    /**
     * 値に該当するラベルを作成する
     * @param keyValue 値
     * @returns ラベル
     */
    public static getLabel(keyValue: number): string {

        switch (keyValue) {
            case ShuushiDocV5YoushikiEdaOutcomeKeihiConstants.JINKENHI:
                return ShuushiDocV5YoushikiEdaOutcomeKeihiConstants.JINKENHI_TEXT;

            case ShuushiDocV5YoushikiEdaOutcomeKeihiConstants.KOUNETSUSUI:
                return ShuushiDocV5YoushikiEdaOutcomeKeihiConstants.KOUNETSUSUI_TEXT;

            case ShuushiDocV5YoushikiEdaOutcomeKeihiConstants.BIHIN_SHOUMOUHIN:
                return ShuushiDocV5YoushikiEdaOutcomeKeihiConstants.BIHIN_SHOUMOUHIN_TEXT;

            case ShuushiDocV5YoushikiEdaOutcomeKeihiConstants.JIMUSHO:
                return ShuushiDocV5YoushikiEdaOutcomeKeihiConstants.JIMUSHO_TEXT;

            default:
                return "";
        }
    }

}
